support/util/performanceprofilestatus/performanceprofilestatus.go

```diff
@@ -0,0 +1,160 @@
+	lastHeartbeatTime := "2024-04-18T06:55:45Z"
+	lastTransitionTime := "2024-04-18T06:55:45Z"
+
+	heartbeatTime, _ := time.Parse(time.RFC3339, lastHeartbeatTime)
+	transitionTime, _ := time.Parse(time.RFC3339, lastTransitionTime)
+
+	conditions := []conditionsv1.Condition{
+		{
+			LastHeartbeatTime:  metav1.Time{Time: heartbeatTime},
+			LastTransitionTime: metav1.Time{Time: transitionTime},
+			Message:            "cgroup=v1;",
+			Status:             corev1.ConditionTrue,
+			Type:               conditionsv1.ConditionAvailable,
+		},
+		{
+			LastHeartbeatTime:  metav1.Time{Time: heartbeatTime},
+			LastTransitionTime: metav1.Time{Time: transitionTime},
+			Status:             corev1.ConditionTrue,
+			Type:               conditionsv1.ConditionUpgradeable,
+		},
+		{
+			LastHeartbeatTime:  metav1.Time{Time: heartbeatTime},
+			LastTransitionTime: metav1.Time{Time: transitionTime},
+			Status:             corev1.ConditionFalse,
+			Type:               conditionsv1.ConditionProgressing,
+		},
+		{
+			LastHeartbeatTime:  metav1.Time{Time: heartbeatTime},
+			LastTransitionTime: metav1.Time{Time: transitionTime},
+			Status:             corev1.ConditionFalse,
+			Type:               conditionsv1.ConditionDegraded,
+		},
+	}
+
+	runtimeClass := "performance-performance"
+	tuned := "openshift-cluster-node-tuning-operator/openshift-node-performance-performance"
+
+	return &performanceprofilev2.PerformanceProfileStatus{
+		Conditions:   conditions,
+		Tuned:        &tuned,
+		RuntimeClass: &runtimeClass,
+	}
+}
+
+func Progressing() *performanceprofilev2.PerformanceProfileStatus {
+	lastTransitionTime := "2024-04-18T06:55:45Z"
+	transitionTime, _ := time.Parse(time.RFC3339, lastTransitionTime)
+
+	conditions := []conditionsv1.Condition{
+		{
+			LastTransitionTime: metav1.Time{Time: transitionTime},
+			Status:             corev1.ConditionFalse,
+			Type:               conditionsv1.ConditionAvailable,
+		},
+		{
+			LastTransitionTime: metav1.Time{Time: transitionTime},
+			Status:             corev1.ConditionFalse,
+			Type:               conditionsv1.ConditionUpgradeable,
+		},
+		{
+			LastTransitionTime: metav1.Time{Time: transitionTime},
+			Status:             corev1.ConditionTrue,
+			Type:               conditionsv1.ConditionProgressing,
+			Reason:             "DeploymentStarting",
+			Message:            "Deployment is starting",
+		},
+		{
+			LastTransitionTime: metav1.Time{Time: transitionTime},
+			Status:             corev1.ConditionFalse,
+			Type:               conditionsv1.ConditionDegraded,
+		},
+	}
+
+	runtimeClass := "performance-performance"
+	tuned := "openshift-cluster-node-tuning-operator/openshift-node-performance-performance"
+
+	return &performanceprofilev2.PerformanceProfileStatus{
+		Conditions:   conditions,
+		RuntimeClass: &runtimeClass,
+		Tuned:        &tuned,
+	}
+}
+
+func Degraded() *performanceprofilev2.PerformanceProfileStatus {
+	lastHeartbeatTime := "2024-04-18T06:55:45Z"
```

Comment:
is this a test stub? if so can we please move into its appropriate _test file?

Reply:
Yes will do so.